pkg/config/app_config.go

```diff
@@ -244,17 +232,41 @@ func (c *AppConfig) GetTempDir() string {
 	return c.TempDir
 }
 
-func configFilePath(filename string) (string, error) {
-	folder, err := findOrCreateConfigDir()
-	if err != nil {
-		return "", err
+func findConfigFile(filename string) (exists bool, path string) {
+	if envConfigDir := os.Getenv("CONFIG_DIR"); envConfigDir != "" {
+		return true, filepath.Join(envConfigDir, filename)
+	}
+
+	legacyConfigPath, err := xdg.SearchConfigFile(filepath.Join("jesseduffield", "lazygit", filename))
+	if err == nil {
+		return true, legacyConfigPath
+	}
+
+	configFilepath, err := xdg.SearchConfigFile(filepath.Join("lazygit", filename))
+	if err == nil {
+		return true, configFilepath
 	}
 
-	return filepath.Join(folder, filename), nil
+	return false, filepath.Join(xdg.ConfigHome, "lazygit", filename)
 }
 
 var ConfigFilename = "config.yml"
 
+func stateFilePath(filename string) (string, error) {
+	stateFile, err := xdg.SearchStateFile(filepath.Join("lazygit", filename))
```

Comment:
Same here: I'd like comments against each step giving an example of which folder it's looking at.

I'm confused that we start by looking for a regular state file, then we fall back to the legacy state file. Shouldn't it be the other way around?

Reply:
added some comments.

> I'm confused that we start by looking for a regular state file, then we fall back to the legacy state file. Shouldn't it be the other way around?

In my mind it made more sense looking for the new path first, now changed to legacy first